addons/website_hr_recruitment/static/tests/tours/website_hr_recruitment.js

```diff
@@ -121,7 +121,8 @@ odoo.define('website_hr_recruitment.tour', function(require) {
         content: 'Verify that the job_id field has kept its default value',
         trigger: 'button[data-action="save"]',
         run: () => {
-            if (!document.querySelector('input[name="job_id"][value="FAKE_JOB_ID_DEFAULT_VAL"]')) {
```

Comment:
Remark for pair-reviewer: Probably I'm missing something here, but normally going to another job application form should apply the `job_id` of the new job and keep it as a default value even in edit mode (which is the main goal of this fix).